src/bio/bio_monitor.c

```diff
@@ -986,19 +999,19 @@ bio_set_vendor_id(struct bio_blobstore *bb, char *bdev_name)
 		return;
 	}
 
-	if (spdk_pci_addr_parse(&pci_addr, binfo.bdi_traddr)) {
+	if (spdk_pci_addr_parse(&opts.pci_addr, binfo.bdi_traddr)) {
 		D_ERROR("Unable to parse PCI address: %s\n", binfo.bdi_traddr);
 		goto free_traddr;
 	}
 
-	for (pci_device = spdk_pci_get_first_device(); pci_device != NULL;
-	     pci_device = spdk_pci_get_next_device(pci_device)) {
-		if (spdk_pci_addr_compare(&pci_addr, &pci_device->addr) == 0) {
-			vid = spdk_pci_device_get_vendor_id(pci_device);
-			break;
-		}
-	}
-	bb->bb_dev_health.bdh_vendor_id = vid;
+	opts.vid = 0;
+
+	spdk_pci_for_each_device(&opts, get_vendor_id);
```

Comment:
Minor: Is there a way to break the loop once the requested entry being processed.

Reply:
as above